algebird-core/src/main/scala/com/twitter/algebird/ScalableBloomFilter.scala

```diff
@@ -0,0 +1,94 @@
+package com.twitter.algebird
+
+import com.googlecode.javaewah.{ EWAHCompressedBitmap => CBitSet }
+import scala.collection.mutable.{ BitSet => MBitSet }
+
+object ScalableBloomFilter {
+
+  /**
+   * The default parameter values for this implementation are based on the findings in "Scalable Bloom Filters",
+   * Almeida, Baquero, et al.: http://gsd.di.uminho.pt/members/cbm/ps/dbloom.pdf
+   */
+  def apply(fpProb: Double,
+    initialCapacity: Int,
+    growthRate: Int = 2,
+    tighteningRatio: Double = 0.9) = {
+    new ScalableBloomFilter(fpProb, initialCapacity, growthRate, tighteningRatio,
+      List(BloomFilter(initialCapacity, fpProb).create()))
+  }
+}
+
+/**
+ * Scalable Bloom Filter - a scalable collection of [[BloomFilterMonoid]] that approximately
+ * maintains its false probability rate as it grows.
+ *
+ * @param fpProb - desired false probability rate
+ * @param headCapacity - expected number of insertions in the initial filter after which a new filter is added
+ * @param growthRate - factor by which each additional bloom filter is larger than the previous
+ * @param tighteningRatio - how much "tighter" the fpProb must be for each successive filter added
+ * @param filters - the underlying [[BF]] collection
+ */
+case class ScalableBloomFilter(fpProb: Double,
+  headCapacity: Int,
+  growthRate: Int,
+  tighteningRatio: Double,
+  filters: List[BF]) extends java.io.Serializable {
+
+  /**
+   * Add a single item to the SBF.  If this increases the estimated number of items
+   * beyond the current filter capacity, a new filter will be created.
+   */
+  def +(item: String) = {
+    val head = filters.head + item
+
+    if (head.size.estimate >= headCapacity) {
```

Comment:
I'm not sure about this check.  It may be more expensive than we need and I'm not even sure it would be correct.

From the paper, it seems that we want a (perhaps protected) method on BF which counts the number of set bits, and we want something like:
```
val initialDistFromOneHalf = math.abs(0.5 - (head.setBits / totalBits))
val newDistFromOneHalf = math.abs(0.5 - (newHead.setBits / totalBits))
if (newDistFromOneHalf > initialDistFromOneHalf) {
  // create new head BF
} else {
  // incorporate into head
}
```

If we wanted to follow the paper even more closely, we'd implement a block bloom filter.  I imagine the math is pretty close either way.